pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"sync"
+	"time"
+)
+
+// ServiceCAMetrics implements metrics gathering for this component.
+type ServiceCAMetrics struct {
+	lock       sync.Mutex
+	expiryTime prometheus.Gauge
+	timeLeft   prometheus.Gauge
+}
+
+func NewServiceCAMetrics() *ServiceCAMetrics {
+	return &ServiceCAMetrics{
+		expiryTime: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "service_ca_expiry_time",
+			Help: "Reports the expiry time of service CA used in the cluster.",
+		}),
+		timeLeft: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "service_ca_time_left",
+			Help: "Reports the time left before the service CA expires.",
+		}),
+	}
+}
+
+func (s *ServiceCAMetrics) Register() {
+	legacyregistry.MustRegister(s)
+}
+
+func (s *ServiceCAMetrics) Create(version *semver.Version) bool {
+	return true
+}
+
+// Describe reports the metadata for metrics to the prometheus collector.
+func (s *ServiceCAMetrics) Describe(ch chan<- *prometheus.Desc) {
+	ch <- s.expiryTime.Desc()
+	ch <- s.timeLeft.Desc()
+}
+
+// Collect calculates metrics from the cached config and reports them to the prometheus collector.
+func (s *ServiceCAMetrics) Collect(ch chan<- prometheus.Metric) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	ch <- s.expiryTime
+	ch <- s.timeLeft
+}
+
+func (s *ServiceCAMetrics) ClearState() {}
+
+func (s *ServiceCAMetrics) Compute(notAfter time.Time) {
```

Comment:
nit: I think `Compute` is too vague a name. Maybe rename to something like `SetCAExpiry`?

Reply:
Yes. Better!